### PR TITLE
Update doc and generic parameter name for JsonValue.GetValue

### PR DESCRIPTION
**Pull request by steveharter**
Adds additional XML doc for the `GetValue` methods; in particular it wasn't obvious that passing in `<object>` for the generic parameter will always succeed and return the underlying value.

In addition, the `JsonNode.GetValue<TValue>` was changed to `JsonNode.GetValue<T>` to be consistent with `JsonValue.TryGetValue<T>`. Not sure on the guidance here, but I think consistency is better, and `<TValue>` seems redundant.

### CODE REVIEW

File: src/libraries/System.Text.Json/ref/System.Text.Json.cs [hunk 1]
```diff
@@ -595,7 +595,7 @@ public abstract partial class JsonNode
         public System.Text.Json.Nodes.JsonObject AsObject() { throw null; }
         public System.Text.Json.Nodes.JsonValue AsValue() { throw null; }
         public string GetPath() { throw null; }
-        public virtual TValue GetValue<TValue>() { throw null; }
+        public virtual T GetValue<T>() { throw null; }
```
Comment (stephentoub) [inline src/libraries/System.Text.Json/ref/System.Text.Json.cs:598]: cc: @terrajobst, @bartonjs

Comment (bartonjs) [reply src/libraries/System.Text.Json/ref/System.Text.Json.cs:598]: Seems fine to me?

Comment (terrajobst) [reply src/libraries/System.Text.Json/ref/System.Text.Json.cs:598]: Same here. I don't think this needs an API review.

Comment (steveharter) [reply src/libraries/System.Text.Json/ref/System.Text.Json.cs:598]: Thanks

File: src/libraries/System.Text.Json/src/System/Text/Json/Nodes/JsonValue.cs [hunk 2]
```diff
@@ -102,10 +102,10 @@ internal override void GetPath(List<string> path, JsonNode? child)
...
-        ///   value supported by current <see cref="JsonElement"/>.<br />
-        ///   Specifying <see cref="object"/> will always succeed and return the underlying value as <see cref="object"/>.<br />
-        ///   The underlying value of a <see cref="JsonValue"/> after deserialization is <see cref="JsonElement"/>,
-        ///   otherwise it is the value specified when the <see cref="JsonValue"/> was created.
+        ///   value supported by current <see cref="JsonElement"/>.
+        ///   Specifying the <see cref="object"/> type for {T} will always succeed and return the underlying value as <see cref="object"/>.<br />
+        ///   The underlying value of a <see cref="JsonValue"/> after deserialization is an instance of <see cref="JsonElement"/>,
+        ///   otherwise it's the value specified when the <see cref="JsonValue"/> was created.
         /// </remarks>
         /// <seealso cref="JsonNode.GetValue{TValue}"></seealso>
```
Comment (eiriktsarpalis) [inline src/libraries/System.Text.Json/src/System/Text/Json/Nodes/JsonValue.cs:110]: Doesn't this also need renaming?
```suggestion
        /// <seealso cref="JsonNode.GetValue{T}"></seealso>
```

**Review by eiriktsarpalis — APPROVED** [review]

File: src/libraries/System.Text.Json/src/System/Text/Json/Nodes/JsonValue.cs (hunk 2 above)
Comment (steveharter) [reply src/libraries/System.Text.Json/src/System/Text/Json/Nodes/JsonValue.cs]: Yep. Thanks

**Pull request merged**